tests/network/localnet/conftest.py

```diff
@@ -38,54 +50,55 @@ def nncp_localnet() -> Generator[libnncp.NodeNetworkConfigurationPolicy]:
         yield nncp
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def namespace_localnet_1() -> Generator[Namespace]:
     yield from create_ns(name="test-localnet-ns1")  # type: ignore
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def namespace_localnet_2() -> Generator[Namespace]:
     yield from create_ns(name="test-localnet-ns2")  # type: ignore
 
 
-@pytest.fixture(scope="module")
-def vlan_id(vlan_index_number: Generator[int]) -> int:
-    return next(vlan_index_number)
-
-
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def nad_localnet_1(
-    namespace_localnet_1: Namespace, vlan_id: int
+    nncp_localnet: Generator[libnncp.NodeNetworkConfigurationPolicy], namespace_localnet_1: Namespace, vlan_id: int
 ) -> Generator[netattachdef.NetworkAttachmentDefinition]:
     with localnet_nad(namespace=namespace_localnet_1.name, name="test-localnet-nad1", vlan_id=vlan_id) as nad:
         yield nad
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def nad_localnet_2(
-    namespace_localnet_2: Namespace, vlan_id: int
+    nncp_localnet: Generator[libnncp.NodeNetworkConfigurationPolicy], namespace_localnet_2: Namespace, vlan_id: int
 ) -> Generator[netattachdef.NetworkAttachmentDefinition]:
     with localnet_nad(namespace=namespace_localnet_2.name, name="test-localnet-nad2", vlan_id=vlan_id) as nad:
         yield nad
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def vm_localnet_1(nad_localnet_1: netattachdef.NetworkAttachmentDefinition) -> Generator[BaseVirtualMachine]:
     with localnet_vm(
-        namespace=nad_localnet_1.namespace, name="test-vm1", network=nad_localnet_1.name, cidr="10.0.0.1/24"
+        namespace=nad_localnet_1.namespace,
+        name="test-vm1",
+        network=nad_localnet_1.name,
+        cidr=f"{IP_NETWORK_PREFIX}.1/24",
```

Comment:
What meaning does `IP_NETWORK_PREFIX` has exactly?
Even if we change the name, then we are still left to define the host value and the mask in a free stile, so the improvement is not that great.

If you want to make sure we get an available address from the same subnet, perhaps something like this will help:
```python
@pytest.fixture
def ipv4_address_pool():
  net_prefix = "10.0.0"
  return (f"{net_prefix}.{host_value}/24" for host_val in range(1, 254))
```

In the future, we can move this to the common lib and consider `net_prefix` and the subnet as optional arguments. But this is not needed now.

If you have this, for each test we will have a new IP pool to work with, looking like this:
`cidr=next(ipv4_address_pool)`